sdk/java-sdk-spring/src/test/java/kalix/spring/testmodels/subscriptions/PubSubTestModels.java

```diff
@@ -122,6 +123,302 @@ public Action.Effect<String> methodTwo(String message) {
+
+    public Action.Effect<Integer> methodOne(Integer message) {
+      return effects().reply(message);
+    }
+
+    public Action.Effect<Integer> methodTwo(Integer message) {
+      return effects().reply(message);
+    }
+  }
+
+  public static class AmbiguousHandlersTopiSubscriptionInAction extends Action {
+
+    @Subscribe.Topic("source")
+    public Action.Effect<Integer> methodOne(Integer message) {
+      return effects().reply(message);
+    }
+
+    @Subscribe.Topic("source")
+    public Action.Effect<Integer> methodTwo(Integer message) {
+      return effects().reply(message);
+    }
+
+    @Subscribe.Topic("source-2")
+    public Action.Effect<Integer> methodThree(Integer message) {
+      return effects().reply(message);
+    }
+  }
+
+  @Subscribe.Topic("source")
+  public static class AmbiguousHandlersTopicTypeLevelSubscriptionInAction extends Action {
+
+    public Action.Effect<Integer> methodOne(Integer message) {
+      return effects().reply(message);
+    }
+
+    public Action.Effect<Integer> methodTwo(Integer message) {
+      return effects().reply(message);
+    }
+  }
+
+  public static class MissingSourceForTopicPublishing extends Action {
+
+    @Publish.Topic("test")
+    public Action.Effect<String> methodOne(String message) {
+      return effects().reply(message);
+    }
+  }
+
+  public static class MissingTopicForVESubscription extends Action {
+
+    @Subscribe.ValueEntity(Counter.class)
+    @Publish.Topic("test")
+    public Action.Effect<String> methodOne(String message) {
+      return effects().reply(message);
+    }
+
+    @Subscribe.ValueEntity(value = Counter.class, handleDeletes = true)
+    public Action.Effect<String> methodTwo() {
+      return effects().ignore();
+    }
+  }
+
+  public static class MissingTopicForESSubscription extends Action {
+
+    @Subscribe.EventSourcedEntity(EmployeeEntity.class)
+    @Publish.Topic("test")
+    public Action.Effect<String> methodOne(String message) {
+      return effects().reply(message);
+    }
+
+    @Subscribe.EventSourcedEntity(EmployeeEntity.class)
+    public Action.Effect<String> methodTwo(Integer message) {
+      return effects().ignore();
+    }
+  }
+
+  @Subscribe.EventSourcedEntity(EmployeeEntity.class)
+  public static class MissingTopicForTypeLevelESSubscription extends Action {
+
+    @Publish.Topic("test")
+    public Action.Effect<String> methodOne(String message) {
+      return effects().reply(message);
+    }
+
+    public Action.Effect<String> methodTwo(Integer message) {
+      return effects().ignore();
+    }
+  }
+
+  public static class MissingTopicForTopicSubscription extends Action {
+
+    @Subscribe.Topic("source")
+    @Publish.Topic("test")
+    public Action.Effect<String> methodOne(String message) {
+      return effects().reply(message);
+    }
+
+    @Subscribe.Topic("source")
+    public Action.Effect<String> methodTwo(Integer message) {
+      return effects().ignore();
+    }
```

Comment:
for more than 1 method subscribing to the same source we will generate the Kalix synthetic method, like
```
rpc KalixSyntheticMethodOnTopicSource(google.protobuf.Any) returns (google.protobuf.Any) {
    option (kalix.method) = {
      eventing {
        in {
           topic: "source"
        }
        out {
           topic: "test"
        }
      }
    };
  }
```
Supporting many synthetic methods for the same source (in one Action) would require more sophisticated logic because the input parameter is always Any 🤯 To achieve what you described you just need split subscriptions to separate Actions.

Reply:
Continuing my thought, once you put a `@Publish.Topic` on a single method, you need to put it on all methods, for the same subscription type.

Reply:
Ah, I see now. Thanks for explaning 👍